packages/uploadthing/src/internal/types.ts

```diff
@@ -25,8 +26,11 @@ export type MaybePromise<TType> = TType | Promise<TType>;
 export type WithRequired<T, K extends keyof T> = T & Required<Pick<T, K>>;
 export type Overwrite<T, U> = Omit<T, keyof U> & U;
 
+// Omitting body because it's not used and is represented
+// by different types in different implementations
+// Anyway, we use json() instead to get the body of the request
 export type RequestLike = Overwrite<
-  WithRequired<Partial<Request>, "json">,
+  Omit<WithRequired<Partial<Request>, "json">, "body">,
   {
     headers: Headers | IncomingHttpHeaders;
```

Comment:
i think it's a bit better ts perf to put this inside the overwrite object (skips 1 Omit)

```ts
export type RequestLike = Overwrite<
  WithRequired<Partial<Request>, "json">,
  {
    body?: never; // we only use `.json`, don't need `body`
    headers: Headers | IncomingHttpHeaders;
  }
>;